Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -0,0 +1,134 @@
+using Robust.Server.Player;
+using Robust.Shared.Player;
+using System;
+using System.Collections.Generic;
+
+namespace Content.Server.PDA.Ringer
+{
+    public class RingerSystem : EntitySystem
+    {
+
+        private readonly Dictionary<string, string> _pathIndex = new Dictionary<string, string>();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            //General Event Subscriptions
+            SubscribeLocalEvent<RingerComponent, ComponentInit>(RandomizeRingtone);
+            //RingerBoundUserInterface Subscriptions
+            SubscribeLocalEvent<RingerComponent, RingerSetRingtoneMessage>(RingerUpdateRingtoneDriver);
+            SubscribeLocalEvent<RingerComponent, RingerPlayRingtoneMessage>(RingerPlayRingtone);
+            SubscribeLocalEvent<RingerComponent, RingerRequestUpdateInterfaceMessage>(UpdateRingerUserInterfaceDriver);
+
+            //Dictionary Path Addition
+            _pathIndex.Add("A","/Audio/Effects/RingtoneNotes/a.ogg");
+            _pathIndex.Add("A#","/Audio/Effects/RingtoneNotes/a#.ogg");
+            _pathIndex.Add("B","/Audio/Effects/RingtoneNotes/b.ogg");
+            _pathIndex.Add("C","/Audio/Effects/RingtoneNotes/c.ogg");
+            _pathIndex.Add("C#","/Audio/Effects/RingtoneNotes/c#.ogg");
+            _pathIndex.Add("D","/Audio/Effects/RingtoneNotes/d.ogg");
+            _pathIndex.Add("D#","/Audio/Effects/RingtoneNotes/d#.ogg");
+            _pathIndex.Add("E","/Audio/Effects/RingtoneNotes/e.ogg");
+            _pathIndex.Add("F","/Audio/Effects/RingtoneNotes/f.ogg");
+            _pathIndex.Add("F#","/Audio/Effects/RingtoneNotes/f#.ogg");
+            _pathIndex.Add("G","/Audio/Effects/RingtoneNotes/g.ogg");
+            _pathIndex.Add("G#","/Audio/Effects/RingtoneNotes/g#.ogg");
+        }
+
+        //Event Functions
+
+        private void RingerPlayRingtone (EntityUid uid, RingerComponent ringer, RingerPlayRingtoneMessage args) {
+            ringer.isPlaying = true; //Allows the overridden Update function to play the ringtone based off of the ringer component's stored ringtonePathArr
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void UpdateRingerUserInterfaceDriver(EntityUid uid, RingerComponent ringer, RingerRequestUpdateInterfaceMessage args) { //Seprate Drivers are needed because UpdateRingerUserInterface() and UpdateRingerComponent() are called by other functions aside from the events themselves
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void RingerUpdateRingtoneDriver(EntityUid uid, RingerComponent ringer, RingerSetRingtoneMessage args) {
+            UpdateRingerComponent(ringer, args.ringtoneStringArgArr);
+        }
+
+        public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, ComponentInit args) { //Randomizes the ringtone
+            string[] possibleNotes = new string[12]{"A","B","C","D","E","F","G","A#","C#","D#","F#","G#"};
```

Comment:
IMO this should just be a field with all 12 notes that can be indexed to get the string representation, + another array field to get the sounds.